[vcpkg] X_VCPKG_APPINSTALL_DEPS_INSTALL optionally install dependencies on install

sandercox opened the pull request:
Continuing on the work from #14129 #13011 this adds an install command override again as proposed initially but behind an `if` check for the variable `X_VCPKG_APPLOCAL_DEPS_INSTALL` marking it as experimental and allowing the feature to _just work_ as I initially intended.

I was already using the install function as used here in my own projects but I think this allows everybody to really use the functionality introduced by the `x_vcpkg_install_local_dependencies` function.

As originally requested in #1653

sandercox committed:
Adding an install command override for dependencies

Shatur reviewed (changes requested):
Works as expected for me with install command and CPack.
One small suggestion: you don't need to expand variables in `if` expressions, this will be done automatically. Elsewhere in this file, the same approach is used, so it's best to do the same for consistency. :) I also tested with this changes, everything works fine for me.

sandercox committed:
Apply suggestions from code review

Use variables directly instead of prefixing with `${}` as is done in rest of the file. Thanks @shatur95

Co-authored-by: Hennadii Chernyshchyk <genaloner@gmail.com>

sandercox committed:
Update scripts/buildsystems/vcpkg.cmake

Co-authored-by: Hennadii Chernyshchyk <genaloner@gmail.com>

JackBoosY commented:
cc @BillyONeal for review this PR.

strega-nil commented:
Given that this is an experimental feature, I'm up for merging it, with the understanding that this very well may be removed or modified, especially as we get closer to stabilization :)
